src/http_server.cc

```diff
@@ -1819,6 +1820,9 @@ HTTPAPIServer::HandleTrace(evhtp_request_t* req, const std::string& model_name)
       req,
       trace_response.AddString("log_frequency", std::to_string(log_frequency)));
   HTTP_RESPOND_IF_ERR(req, trace_response.AddString("trace_file", filepath));
+  HTTP_RESPOND_IF_ERR(
+      req, trace_response.AddString(
+               "trace_mode", TRITONSERVER_InferenceTraceModeString(mode)));
```

Comment:
Wouldn’t the “mode” be the property of the trace manager created? I don’t think it requires Triton core changes, and the change in core repo doesn’t do anything special with respect to the mode given.

Reply:
I thought that it would make sense to put `mode` into core now, since later implementations may touch the core as well. For example, `infer_trace.h` has `Report` to report trace activity. I thought it would make sense to later incorporate opentelemetry's span updates there. Plus, we would also need to add `bls` support for opentelemetry. That being said, I understand that the more discussions are need to happen around the implementation, but I believe it make sense to keep `mode` changes in the core repo for now. Let me know if I am wrong.

Reply:
I think the core changes in `tritonserver.h` etc. are not necessary right now. If we want them later, we can add them. But if we add them now and they're not needed, we have to keep it in there for backwards compatibility. So I think it's better to not add it until we really need it, for any C APIs.

Reply:
I don't think `TRITONSERVER_InferenceTraceLevelString` is currently used in any of our repos. It may be used by custom in-process applications written by users, but it's likely not a critical thing to have until requested.

Similarly, it may make more sense to pivot towards some `TRITONSERVER_InferenceTraceConfig()` with the move to `--trace-config` rather than exposing a new API for each field, but again probably not needed until requested.

Reply:
Got it, I'll incorporate changes regarding the `mode`